src/inmanta/deploy/scheduler.py

```diff
@@ -433,6 +442,50 @@ async def _run_for_agent(self, agent: str) -> None:
 
             self._work.agent_queues.task_done(agent, task)
 
+    async def _start_compliance_check(self, resource: ResourceIdStr) -> None:
+        """
+        This method is expected to be called when the given resource has just finished deploying,
+        no matter the result itself i.e. success or failure.
+        """
+        if (previous_task := self.resource_periodic_deploys.get(resource)) is not None:
+            previous_task.cancel()
+
+        self.resource_periodic_deploys[resource] = self.create_recurring_compliance_check(resource)
+
+    def create_recurring_compliance_check(self, resource: ResourceIdStr) -> asyncio.Task[None]:
+        """
+        Schedule recurring deploys for the given resource every <scheduler_resource_compliance_check_window> seconds
+        and return the associated task. This task will be canceled and scheduled again when the next deploy finishes
+        for this resource.
+        """
+
+        async def delay(coro: Coroutine[Any, Any, None], seconds: float) -> None:
+            """
+            Helper function to run a coroutine after a delay
+            """
+            await asyncio.sleep(seconds)
```

Comment:
This delay mechanism was not what I had in mind when thinking about timers. 

I can't determine if it is very smart or very dangerous.

I am suspicious about it and I wonder why we don't use any of the existing scheduling tools we have.

Reply:
This is how I interpreted this part of the ticket `Using the eventloop timer`. I'll go with the scheduling framework instead.